static/js/lib/fetch.js

```diff
@@ -0,0 +1,381 @@
+(function() {
```

Comment:
I copied it in from bower locally since we didn't have anything setup for managing front end dependencies. We should have something though.

I totally forgot it used es6 promises. My rationale was I use it for everything now and didn't see requests. Your right that I should 
 use jQuery instead though. I'll switch it out.